addons/website/static/src/components/dialog/dialog.js

```diff
@@ -124,57 +116,406 @@ export class AddPageDialog extends Component {
                 'view_mode': 'form',
             });
         } else {
-            this.website.goToWebsite({path: data.url, edition: true, ...(this.props.selectWebsite && {websiteId})});
+            this.website.goToWebsite({path: data.url, edition: true, websiteId: this.props.websiteId});
         }
         this.props.onAddPage(this.state);
     }
 }
-AddPageDialog.props = {
+AddPageConfirmDialog.props = {
     close: Function,
     onAddPage: {
         type: Function,
         optional: true,
     },
-    selectWebsite: {
+    websiteId: Number,
+    sectionsArch: String,
+    name: String,
+};
+AddPageConfirmDialog.defaultProps = {
+    onAddPage: NO_OP,
+};
+AddPageConfirmDialog.components = {
+    Switch,
+    WebsiteDialog,
+};
+AddPageConfirmDialog.template = "website.AddPageConfirmDialog";
+
+export class AddPageTemplatePreview extends Component {
+    setup() {
+        super.setup();
+        this.iframeRef = useRef("iframe");
+        this.previewRef = useRef("preview");
+        this.holderRef = useRef("holder");
+
+        onMounted(async () => {
+            const holderEl = this.holderRef.el;
+            if (this.props.isBlank) {
+                holderEl.classList.add("o_ready");
+                return;
+            }
+            holderEl.classList.add("o_loading");
+            if (this.props.isPreloading) {
+                return;
+            }
+            const previewEl = this.previewRef.el;
+            const iframeEl = this.iframeRef.el;
+            // Firefox replaces the built content with about:blank.
+            const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+            if (isFirefox) {
+                // Make sure empty preview iframe is loaded.
+                // This event is never triggered on Chrome.
+                await new Promise(resolve => {
+                    iframeEl.contentDocument.body.onload = resolve;
+                });
+            }
+            // Apply styles.
+            for (const cssLinkEl of await this.props.getCssLinkEls()) {
```

Comment:
:thinking:  This would also make it required to flush this cache whenever the theme or the selected website is changed.

@detrouxdev Maybe you know of an owl mechanism that would be adequate for this ?

Reply:
I'll just quickly write my reply here so that it's also written somewhere but I see two ways for this:

- Make the getCssLinkEls accessible through the env using `useSubEnv` so that any child component can have access to it. It avoids having to propagate it through props.

- If we want the cache to persists, then I would recommend using the website service, as we can access it through legacy widgets now (thanks to [this](https://github.com/odoo/odoo/blob/7470797be7b4b23a433d3fa87e014b8f8b1f364d/addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js#L788) and [this](https://github.com/odoo/odoo/blob/7470797be7b4b23a433d3fa87e014b8f8b1f364d/addons/web/static/src/legacy/js/core/service_mixins.js#L63) ) you can just do `this.website.call("website", "myMethod")` only methods though. You could have a method to get the cache and a method to clear it.

I think the first solution is best for now though! 😄

Reply:
Replaced all those props by `useSubEnv`. Thanks.